### ISSUE TITLE
Not good enough search results

### ISSUE THREAD

**Issue by KronosTheLate** [typst/typst#910]
Below is a list of search words (with actual results) that I expected to return functions that make a string uppercase or lowercase (so the functions `upper` and `lower`), but that do not return these functions:
![image](https://user-images.githubusercontent.com/61620837/233581695-6bec4501-ccf5-4c22-8d0a-c1c610feb5a4.png)
![image](https://user-images.githubusercontent.com/61620837/233581771-6af90e77-6f8c-46bb-af5f-384ff0d360ae.png)
![image](https://user-images.githubusercontent.com/61620837/233581831-efd1a80f-5667-469d-b60c-05b6bb0f5d43.png)
![image](https://user-images.githubusercontent.com/61620837/233581902-a62f74b8-1577-4a19-8e47-4e5f20b080d8.png)

It is not at all obvious that one should go looking for the abbreviated forms `lower` and `upper`. Having performed the searches, and having looked through the [doc page on strings](https://typst.app/docs/reference/types/string/), I concluded that no such functions exist, and made my own version with regex matching.

The issue then is that the existing functions are not discoverable enough, as I feel that my search for them is about the maximum amount that can be expected of new users, and I could not find them. I do now know if one would have similar problems with other specific searches, but having searched the docs a few times now, I am left with a general feeling that the hits are not great. Often I have to try multiple searches with different formulations, as opposed to what I find most obvious.

A possible solution would be to add a keywords section on the different pages to ensure that related keywords bring up the page.

**Comment by laurmaedje** [comment]
The search indeed isn't great, it could benefit from keywords and a bit better matching algorithm. We'll improve it in time.

I'm a bit confused by the first two screenshots though as they _do_ show `lower` and `upper` as the top hit. They just show the display name (which is Lowercase / Uppercase).

**Comment by KronosTheLate** [comment]
Oh snap, my bad! That kind of takes the wind out of the sails of this issue... I am not sure at what point I have brainfarted in my inability to find the function then.

So if the correct functions are returned, and the general issue is on the agenda, there is no point in this one staying open.

As a sidenote, I want to mention that I find the displayname and actual function name distinction to be somewhat confusing. Perhaps aliases for the same function could be another approach? So that `lower(args)` just calls `lowercase(args)`, with the underlying function having long descriptive names, and some of them getting convenient aliases on top. But perhaps this does not solve the problem that the displayname/functionname distinction originally tries to solve - thoughts?

**Comment by KronosTheLate** [comment]
Come to think of it, I believe this is what happened:
I tried to use a non-existing function `uppercase`, which did not work. When it did not, I searched the docs. So the hit on some `uppercase` function can not possibly be what I am looking for, because no such function exists!

So this displayname/functionname distrinction is I think what caused my confusion. Should I make a new issue on that, or can someone put that point to rest immediately with some good reason why it should be that way?
